### PR TITLE
fixes stale session bug

### PR DESCRIPTION
**Pull request by aesnyder**

### CODE REVIEW

File: app/views/layouts/application.html.slim [hunk 1]
```diff
@@ -6,18 +6,18 @@ html
...
+    .navbar.navbar-default.header-wrapper
+      .container-fluid
+        .navbar-header
           h1 = link_to 'Hive', root_path
           .breadcrumbs
             = render_breadcrumbs separator: ' / '
-        .log-in
-          - if current_user
-
-              => current_user.email
```
Comment (DavidDufresne) [inline app/views/layouts/application.html.slim]: Why was this removed?

Comment (aesnyder) [inline app/views/layouts/application.html.slim]: Why not?

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -6,7 +6,13 @@ class ApplicationController < ActionController::Base
 
   private
 
+  def ensure_user
+    redirect_to '/auth/github' unless current_user.present?
+  end
+
   def current_user
-    @current_user ||= User.find(session[:user_id]) if session[:user_id]
+    if session[:user_id]
```
Comment (DavidDufresne) [inline app/controllers/application_controller.rb]: I'd probably go with
```
@current_user ||= begin
  User.find(session[:user_id]) if session[:user_id]
end
```

Comment (aesnyder) [inline app/controllers/application_controller.rb]: I'm not sure I find that easier to read, do you @joelhawksley?

Comment (DavidDufresne) [inline app/controllers/application_controller.rb]: You save having to check the session variable.  Or is that part of the bug fix?
